### PR TITLE
Documentation: Fixed spelling mistakes and formatting of .md files.

### PR DESCRIPTION
**Pull request by Peredurz**
I saw that there were a few spelling mistakes in the Documentation directory markdown files.
I have now fixed these.

There were also a few documents where the formatting was incorrect, so with the help of ltex, a markdown plugin for neovim, I have hopefully fixed that as well.

### CODE REVIEW

File: Documentation/Patterns.md [hunk 1]
```diff
@@ -223,7 +224,7 @@ TEST_CASE(string_view_literal_operator)
 ## Source Location
 
 C++20 added std::source_location, which lets you capture the
-callers __FILE__ / __LINE__ / __FUNCTION__ etc as a default
+callers **FILE** / **LINE** / **FUNCTION** etc. as a default
```
Comment (AtkinsSJ) [inline Documentation/Patterns.md:227]: This is wrong, the variables are named `__FILE__`, `__LINE__` and `__SOURCE__`, it's not formatting. But maybe they should have ` around them.

File: Documentation/NvimConfiguration.md [hunk 2]
```diff
@@ -96,22 +97,27 @@ To change the coc-settings.json go to the file `~/.config/nvim/coc-settings.json
...
-`coc-settings.json` you might want to nuke it first and
-work towards your desired config by adding the other parts back in to avoid
-conflicts.
+> `coc-settings.json` you might want to nuke it first and
+> work towards your desired config by adding the other parts back in to avoid
+> conflicts.
 
-> **Note**: If you have configured `clangd` as a languageServer in
-`coc-settings.json`, you should remove it to avoid running clangd twice!
+> **Note**: If you have configured `clangd` as a language server in
```
Comment (AtkinsSJ) [inline Documentation/NvimConfiguration.md:104]: `languageServer` is also a name, and shouldn't be changed.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Hi there!

There are some good fixes in here, but a lot of it feels a bit arbitrary. Using asterisks for italics, or for bullet points, and having a single space between a bullet and its content, are all perfectly valid Markdown, and it doesn't feel useful to conform to a particular style unless we're going to enforce that.

I did spot a couple of places where you've accidentally made the information wrong, and I've marked them. I haven't looked through this thoroughly though.

Oh also, the commit title should be an imperative, so "Fix spelling..." not "Fixed spelling..."

File: Documentation/Patterns.md (hunk 1 above)
Comment (Peredurz) [reply Documentation/Patterns.md:227]: Ok, I'll try to fix these issues, I have noticed that when I saved the files that it automatically changed these things to asterisks. I'll have to look into my own lsp settings for that.

**Comment by TobyAsE** [comment]
Hi there!

The PR contains the same commit twice (same message, but the second has no change set) and the third commit changes stuff from the first commit.

Please use `git rebase` and fixup to update the initial changes you made, then force-push the updates.

**Comment by Peredurz** [comment]
I think I've done everything correctly now, I changed the title of the initial commit and combined the later changes into one commit. Is this how it should be done?

**Pull request closed without merging**